lib/friends.rb

```diff
@@ -3,5 +3,5 @@
 require "friends/version"
 
 module Friends
-  TAG_REGEX = /@\p{Alnum}+/
+  TAG_REGEX = /(?<!\p{Alnum})@\p{Alnum}(?:[:-]?\p{Alnum})*/
```

Comment:
One small thing I'm wondering here though... is it weird that tags match case-sensitively? Should `@games:caverna` and `@games:Caverna` really be different?

(To be clear, I had been wondering this already—it's nothing to do with your changes—and I'm happy to make that change in a separate PR. More just curious for your thoughts.)

Reply:
Exact-byte-match (bolstered by a strong convention of all-lowercase tags) is certainly the _easiest_ thing to do, and more complicated things aren't a [Pareto improvement](https://en.wikipedia.org/wiki/Pareto_efficiency) (an improvement that makes at least one person better off and nobody worse off). I could imagine someone using @‌nice for "this was a nice thing that happened" and @‌Nice for "something concerning this one city in the southeast of France", or something else that differs only by capitalization.

I'd say "don't do any of this until someone asks for it", but it's worth thinking about these things as a group, although implementing one doesn't necessarily mean you're obliged to implement them all:

- case-insensitive tag matching (but case-preserving, like HFS+, APFS, and NTFS file names)
- ensuring that all strings are folded into one particular normalization form (Ruby might do this properly itself; I don't know.)
- diacritic-insensitive matching, at least for some locales

Nobody's asking for any of this AFAIK, so I'd personally hesitate to implement any of these features if `friends` were my project.

Reply:
Wow, thanks for such a thoughtful response. Yeah I guess I buy the argument that it's fine to leave these for now until someone needs them to behave otherwise.